core/trino-main/src/main/java/io/trino/execution/scheduler/FaultTolerantPartitioningSchemeFactory.java

```diff
@@ -63,6 +68,12 @@ public FaultTolerantPartitioningScheme get(PartitioningHandle handle, Optional<I
         return result;
     }
 
+    public void disableCaching()
```

Comment:
Do we need to disable? There's a chance that a different partitioning scheme may get created for the same partitioning handle. 

It doesn't matter for system `HASH` distribution, but it does for bucketed tables. 

For example different `FaultTolerantPartitioningScheme#partitionToNodeMap` may get assigned for two sides of the same join what may result in incorrect results. 


Should we simply reset instead? Is it safe to reset?

Reply:
Can't simply reset.

I changed to use the supplied `partitionCount` as the source of truth (if it's present). Other part still use the caching behavior.